src/WakuRlnV2.sol

```diff
@@ -112,14 +120,39 @@ contract WakuRlnV2 {
         idCommitmentIndex += 1;
     }
 
+    /// @notice Checks if a commitment is valid
+    /// @param idCommitment The idCommitment of the member
+    /// @return true if the commitment is valid, false otherwise
     function isValidCommitment(uint256 idCommitment) public pure returns (bool) {
         return idCommitment != 0 && idCommitment < Q;
     }
 
+    /// @notice Checks if a user message limit is valid
+    /// @param userMessageLimit The user message limit
+    /// @return true if the user message limit is valid, false otherwise
+    function isValidUserMessageLimit(uint32 userMessageLimit) public view returns (bool) {
+        return userMessageLimit > 0 && userMessageLimit <= MAX_MESSAGE_LIMIT;
+    }
+
+    /// @notice Returns the rateCommitment of a member
+    /// @param index The index of the member
+    /// @return The rateCommitment of the member
```

Comment:
missing a test for `indexToCommitment`? Should we check as well that `index` is within bounds? inside `idCommitmentIndex`.

as it is i guess `index` can be greater than `idCommitmentIndex` and IIRC that will "cycle" back to hashed elements. i mean, imagine a tree with 8 leafs. iirc the element 9 will be the hash of leafs 0-1, element 10 based 2-3, etc.

Reply:
addressed in dda2aff
refactored to make this function internal, we only want users to call `idCommitmentToMetadata`, which protects against this
and general practice is not to add reverts/checks to view functions :)